### PR TITLE
Add `public` statement to `Base.GC`

### PR DESCRIPTION
**Pull request by lgoettgens**
https://github.com/JuliaLang/julia/issues/51335 for the `Base.GC` submodule (cc @LilithHafner)

This PR marks the following symbols as public:
```julia
gc
enable
enable_finalizers
in_finalizer
@preserve
safepoint
enable_logging
logging_enabled
```

All of these have a docstring and seem to be useful for external code.

This PR should be backported to 1.11.

### CODE REVIEW

**Comment by PallHaraldsson** [comment]
Maybe skip:
help?> Base.GC.enable_finalizers

and possibly GC.in_finalizer and all related to finalizers? I see finalizers are deprecated in Java, and I didn't even know of those... If it makes sense to drop them, then not expand the API? Are they used in some packages? I actually thought the rest e.g. GC.gc was part of the API already, not objecting to it public, though I'm not sure anyone should really be using it...

**Comment by giordano** [comment]
I don't remember having ever talked about finalisers and GC :flushed:

**Comment by gbaraldi** [comment]
It was me :).

File: base/gcutils.jl [hunk 1]
```diff
@@ -109,6 +109,8 @@ Module with garbage collection utilities.
 """
 module GC
 
+public gc, enable, @preserve, safepoint, enable_logging, logging_enabled
```
Comment (vchuravy) [inline base/gcutils.jl:112]: I think `enable_logging` and `logging_enabled` should maybe not have been made public?

I understand public to be part of the API guarantee?

**Pull request merged**